display: sdl: add CONFIG_SDL_DISPLAY_COLOR_TINT

Finomnis opened the pull request:
Allows the SDL display to be tinted in a specific color, to allow the simulation of monochrome colored displays like `R8` (`L8` + tint `0xff0000`).

----

LVGL widgets demo in `L8` colorspace:

```
west build -b native_sim/native/64 samples/modules/lvgl/demos -t run -p auto -- \
    -DCONFIG_LV_Z_DEMO_WIDGETS=y \
    -DCONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_L_8=y
```

<img width="488" height="302" alt="image" src="https://github.com/user-attachments/assets/91e3e02d-2dd6-4e1f-be35-814cb8664c2e" />

-----

LVGL widgets demo in `L8` colorspace with tint `0xff0000`, simulating an `R8` display:

```
west build -b native_sim/native/64 samples/modules/lvgl/demos -t run -p auto -- \
    -DCONFIG_LV_Z_DEMO_WIDGETS=y \
    -DCONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_L_8=y \
    -DCONFIG_SDL_DISPLAY_COLOR_TINT=0xff0000
```

<img width="486" height="300" alt="image" src="https://github.com/user-attachments/assets/1a79ff84-9655-44a7-b6d1-5685d017f8ea" />


----

Or, in combination with #94428:

```
west build -b native_sim/native/64 samples/modules/lvgl/screen_transparency -t run -p auto -- \
    -DCONFIG_SDL_DISPLAY_DEFAULT_PIXEL_FORMAT_AL_88=y \
    -DCONFIG_SDL_DISPLAY_COLOR_TINT=0xff0000
```

<img width="329" height="269" alt="image" src="https://github.com/user-attachments/assets/6a69ef5c-f49a-4f11-a8c1-028ed5a2ed94" />

Finomnis committed:
display: sdl: add CONFIG_SDL_DISPLAY_COLOR_TINT

Allows the SDL display to be tinted in a specific color,
to allow the simulation of monochrome colored displays
like `R8` (`L8` + tint `0xff0000`)

Signed-off-by: Martin Stumpf <finomnis@gmail.com>

kartben commented on drivers/display/Kconfig.sdl:
```diff
@@ -88,6 +88,12 @@ config SDL_DISPLAY_TRANSPARENCY_GRID_CELL_COLOR_2
 	help
 	  The color of the even cells in the transparency grid.
 
+config SDL_DISPLAY_COLOR_TINT
```
new Kconfig needs to be mentioned in release notes

Finomnis replied:
Like so?

Finomnis commented on drivers/display/display_sdl_bottom.c:
```diff
@@ -130,7 +130,12 @@ void sdl_display_write_bottom(const uint16_t height, const uint16_t width, const
 	if (display_on && !frame_incomplete) {
 		SDL_RenderClear(renderer);
 		SDL_RenderCopy(renderer, background_texture, NULL, NULL);
+		SDL_SetTextureColorMod(texture,
```
I don't think it makes much of a difference tbh

faxe1008 commented on drivers/display/Kconfig.sdl:
```diff
@@ -88,6 +88,12 @@ config SDL_DISPLAY_TRANSPARENCY_GRID_CELL_COLOR_2
+	hex "Color filter applied on top of the display data"
+	default 0xffffff
+	help
+	  Can be used to simulate colored monochrome displays, like R8 (L8 + tint 0xff0000).
```
jep, looks good

aescolar commented:
Changing assignees as the change in the release notes has been reviewed and it is a trivial change compared to the rest.

Finomnis commented:
Rebased and fixed merge conflict. Also, minor beautification in docs (suggested here https://github.com/zephyrproject-rtos/zephyr/pull/94428#discussion_r2277013926)

aescolar commented:
> Unsure why twister failed. Server issues?

So it seems. Retriggered
